Windows support?

When using `cy2` on Windows I get an error:

```
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'D:\project\src\lib\tasks\state'
Require stack:
- D:\project\src\node_modules\cy2\lib\patch.js
- D:\project\src\node_modules\cy2\bin\cy2
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at D:\project\src\node_modules\cy2\lib\patch.js:19:19
    at D:\project\src\node_modules\npm-which\index.js:35:9
    at D:\project\src\node_modules\which\which.js:87:20
    at D:\project\src\node_modules\isexe\index.js:42:5
    at D:\project\src\node_modules\isexe\windows.js:36:5
    at FSReqCallback.oncomplete (fs.js:184:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\project\\src\\node_modules\\cy2\\lib\\patch.js',
    'D:\\project\\src\\node_modules\\cy2\\bin\\cy2'
  ]
}
```

--- a/README.md
+++ b/README.md
@@ -22,3 +22,53 @@ CYPRESS_API_URL="https://sorry-cypress.domain.com" cy2 run  --parallel --record 
 ```
 
 When `CYPRESS_API_URL` is not available, it just uses the default API server.
+
+## API
+
+Patch `cypress` package
+
+```ts
+patch(apiURL: string) => Promise<void>
+```
+
+Path and run `cypress`
+
+```ts
+run(apiURL?: string = 'https://yourserver.io/'), label?: string = 'cy2')=> Promise<void>
+```
+
+### Examples
+
+```js
+const { patch } = require('cy2');
+
+async function main() {
+  await patch('https://sorry-cypress-demo-director.herokuapp.com');
+}
+
+main().catch(console.error);
+```
+
+```js
+#!/usr/bin/env node
+
+/* cmd.js */
+
+
+const { run } = require('cy2');
+
+async function main() {
+  await run('https://sorry-cypress-demo-director.herokuapp.com/', 'myCMD');
+}
+
+main().catch(console.error);
+/*
+
+$ ./cmd.js --help
+[myCMD] Running cypress with API URL: https://sorry-cypress-demo-director.herokuapp.com/
+Usage: cypress <command> [options]
+
+Options:
+  -v, --version      prints Cypress version
+  -h, --help         display help for command
+```

--- a/index.js
+++ b/index.js
@@ -1,13 +1,11 @@
 const lib = require('./lib/patch');
 
-exports.patch = async (cypressAPIUrl) => {
-  await lib.patch(cypressAPIUrl, await lib.getCypressCLIBinPath());
-};
+const DEFAULT_OVERRIDE_URL = 'https://api.cypress.io/';
 
-exports.run = async (cypressAPIUrl, label = 'cy2') => {
-  const cliBinPath = await lib.getCypressCLIBinPath();
-  lib.patch(cypressAPIUrl, cliBinPath);
-  console.log(`[${label}] Launching cypress from ${cliBinPath}`);
-  console.log(`[${label}] Cypress API URL: ${cypressAPIUrl}`);
-  lib.run(cliBinPath);
+exports.patch = lib.patch;
+
+exports.run = async (cypressAPIUrl = DEFAULT_OVERRIDE_URL, label = 'cy2') => {
+  await lib.patch(cypressAPIUrl);
+  console.log(`[${label}] Running cypress with API URL: ${cypressAPIUrl}`);
+  await lib.run();
 };

--- a/lib/patch.js
+++ b/lib/patch.js
@@ -4,25 +4,11 @@ const cp = require('child_process');
 const which = require('npm-which')(process.cwd());
 const yaml = require('js-yaml');
 
-const DEFAULT_OVERRIDE_URL = 'https://api.cypress.io/';
-
-exports.getCypressCLIBinPath = function getCypressCLIBinPath() {
-  return new Promise((resolve, reject) => {
-    which('cypress', function (err, binaryCypress) {
-      if (err) {
-        return reject(err.message);
-      }
-      return resolve(fs.realpathSync(binaryCypress));
-    });
-  });
-};
-
-exports.getConfigFilesPaths = getConfigFilesPaths;
-
-exports.patch = async function (apiURL, cliBinPath) {
-  const { configFilePath, backupConfigFilePath } = getConfigFilesPaths(
-    cliBinPath
-  );
+exports.patch = async function (apiURL) {
+  if (!apiURL) {
+    throw new Error('Missing apiURL');
+  }
+  const { configFilePath, backupConfigFilePath } = await getConfigFilesPaths();
   try {
     fs.statSync(backupConfigFilePath);
   } catch (e) {
@@ -30,32 +16,53 @@ exports.patch = async function (apiURL, cliBinPath) {
   }
 
   const doc = yaml.load(fs.readFileSync(configFilePath, 'utf8'));
-  doc.production.api_url = apiURL || DEFAULT_OVERRIDE_URL;
+  doc.production.api_url = apiURL;
 
   fs.writeFileSync(configFilePath, yaml.dump(doc));
 };
 
-exports.run = async function (cliBinPath) {
+exports.run = async function () {
   const [node, script, ...rest] = process.argv;
+  const cliBinPath = await getCypressCLIBinPath();
   const child = cp.spawn(cliBinPath, [...rest], { stdio: 'inherit' });
   child.on('exit', (code) => process.exit(code));
 };
 
-function getConfigFilesPaths(cliBinPath) {
-  const cliBinPathElements = cliBinPath.split('/');
-  const cliPkgPath = cliBinPathElements
-    .slice(0, cliBinPathElements.length - 2)
-    .join('/');
+async function getCypressRoot() {
+  let cypressRoot;
+  try {
+    const cliBinPath = await getCypressCLIBinPath();
+    const cliBinPathElements = cliBinPath.split(path.sep);
 
-  const stateModulePath = path.resolve(cliPkgPath, 'lib/tasks/state');
+    if (isWindowsBin(cliBinPath)) {
+      cypressRoot = path.resolve(
+        path.dirname(cliBinPath),
+        'node_modules',
+        'cypress'
+      );
+    } else {
+      cypressRoot = cliBinPathElements
+        .slice(0, cliBinPathElements.length - 2)
+        .join('/');
+    }
+    return cypressRoot;
+  } catch (error) {
+    throw new Error('Cannot detect cypress. Is cypress installed?');
+  }
+}
+async function getConfigFilesPaths() {
+  const cypressRoot = await getCypressRoot();
+  const stateModulePath = path.resolve(cypressRoot, 'lib/tasks/state');
   const state = require(stateModulePath);
-  const pkgFilePath = state.getBinaryPkgPath(state.getBinaryDir());
+
+  const pkgPath = state.getBinaryPkgPath(state.getBinaryDir());
+  const pkgRoot = path.dirname(pkgPath);
   const configFilePath = path.resolve(
-    path.dirname(pkgFilePath),
+    pkgRoot,
     'packages/server/config/app.yml'
   );
   const backupConfigFilePath = path.resolve(
-    path.dirname(pkgFilePath),
+    pkgRoot,
     'packages/server/config/_app.yml'
   );
 
@@ -63,4 +70,19 @@ function getConfigFilesPaths(cliBinPath) {
     configFilePath,
     backupConfigFilePath,
   };
+}
+
+function getCypressCLIBinPath() {
+  return new Promise((resolve, reject) => {
+    which('cypress', function (err, binaryCypress) {
+      if (err) {
+        return reject(err.message);
+      }
+      return resolve(path.normalize(fs.realpathSync(binaryCypress)));
+    });
+  });
+}
+
+function isWindowsBin(path) {
+  return path.includes('.CMD');
 }

--- a/package.json
+++ b/package.json
@@ -23,9 +23,6 @@
   "engines": {
     "node": ">=10.0.0"
   },
-  "peerDependencies": {
-    "cypress": "*"
-  },
   "keywords": [
     "cypress",
     "wrapper",

